### PR TITLE
[Disability Benefits Team 1] Track service provider for Form526 Submissions 

### PR DESCRIPTION
**Pull request by tommasina-va**
This PR: 

- adds in service provider as a string (either 'lighthouse' or 'evss') to the `system_transaction_id` used in logs internally and for Lighthouse's internal use. 
- adds service_provider to `with_tracking` for Rails logger and DataDog's StatsD metrics. 

## Summary

- *This work is behind a feature toggle (flipper): YES/NO* NO
- *(Summarize the changes that have been made to the platform)*
- *(If bug, how to reproduce)*
- *(What is the solution, why is this the solution?)*
- *(Which team do you work for, does your team own the maintenance of this component?)*
- *(If introducing a flipper, what is the success criteria being targeted?)*

## Related issue(s)

- *Link to ticket created in va.gov-team repo OR screenshot of Jira ticket if your team uses Jira*  https://github.com/department-of-veterans-affairs/va.gov-team/issues/93563
## Testing done

- [x] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by aurora-a-k-a-lightning — APPROVED** [review]

**Review by stevenjcumming — CHANGES_REQUESTED** [review]
Please resolve the `MethodLength` cop

File: app/models/form526_submission.rb [hunk 1]
```diff
@@ -82,7 +82,8 @@ class Form526Submission < ApplicationRecord
   # example: can be used as a search parameter in Datadog
   # TODO: follow-up in ticket #93563 to make this more robust, i.e. attempts of jobs, etc.
   def system_transaction_id
-    "Form526Submission_#{id}"
+    service_provider = saved_claim.parsed_form['startedFormVersion'].present? ? 'lighthouse' : 'evss'
+    "Form526Submission_#{id}, user_uuid: #{User.find(user_uuid)}, service_provider: #{service_provider}"
```
Comment (freeheeling) [inline app/models/form526_submission.rb:86]: Do you want the entire `User` object or only the `user_uuid`?

**Review by freeheeling — COMMENTED** [review]
Question regarding `system_tracking_id` update.

File: app/models/form526_submission.rb (hunk 1 above)
Comment (tommasina-va) [reply app/models/form526_submission.rb:86]: good catch :) thanks!

**Review by aurora-a-k-a-lightning — APPROVED** [review]

**Comment by tommasina-va** [comment]
> Please resolve the `MethodLength` cop

@stevenjcumming Done. Thanks!

**Review by stevenjcumming — APPROVED** [review]

**Pull request merged**